packages/block-editor/src/components/colors/README.md

```diff
@@ -0,0 +1,76 @@
+# Color HOC and Hook
+
+Set of functions to enable color functionality in Blocks.
+
+## `withColors` HOC
+
+A higher-order component, which handles color logic for class generation color value, retrieval and color attribute setting.
+
+### Usage
+
+```jsx
+export default compose(
+	withColors( 'backgroundColor', { textColor: 'color' } ),
+	MyColorfulComponent
+);
+```
+
+## `createCustomColors` HOC
+
+A higher-order component factory for creating a 'withCustomColors' HOC, which handles color logic for class generation color value, retrieval and color attribute setting. Use this higher-order component to work with a custom set of colors.
+
+### Usage
+
+```jsx
+import { createCustomColorsHOC } from '@wordpress/block-editor';
+
+const CUSTOM_COLORS = [
+	{ name: 'Red', slug: 'red', color: '#ff0000' },
+	{ name: 'Blue', slug: 'blue', color: '#0000ff' },
+];
+const withCustomColors = createCustomColorsHOC( CUSTOM_COLORS );
+
+export default compose(
+	withCustomColors( 'backgroundColor', 'borderColor' ),
+	MyColorfulComponent
+);
+```
+
+## `useColors` hook
+
+Experimental hook that combines the functionality of the `withColors` HOC and `PanelColorSettings`, with `InspectorControls` into an easy to use `useColors` hook.
```

Comment:
I guess instead of "Experimental hook that combines the functionality of the `withColors` HOC and `PanelColorSettings`, with `InspectorControls` into an easy to use `useColors` hook.".

We can just say that it provides equivalent functionality to `withColors` as a react hook.

Reply:
I've modified the `useColors` explanation as you suggested.